test/PlatformFunctions.spec.ts

```diff
@@ -5,12 +5,12 @@ require('./testUtils.js');
 import * as fs from 'fs';
 import * as path from 'path';
 
-import {FetchHttpClient} from '../lib/net/FetchHttpClient';
-import {NodeCryptoProvider} from '../lib/crypto/NodeCryptoProvider';
-import {NodeHttpClient} from '../lib/net/NodeHttpClient';
-import {NodePlatformFunctions} from '../lib/platform/NodePlatformFunctions';
-import {PlatformFunctions} from '../lib/platform/PlatformFunctions';
-import {SubtleCryptoProvider} from '../lib/crypto/SubtleCryptoProvider';
+import {FetchHttpClient} from '../cjs/net/FetchHttpClient';
```

Comment:
Do we want to duplicate the test suite for `esm/` files? I haven't done this yet since this would double the size of our tests when the underlying functionality is the same, and the main issues we're worried about with the additional ESM entry point should be captured in integration tests (which I'll add to as needed).

Reply:
> Do we want to duplicate the test suite for esm/ files? 

No. For the exact reason you outlined.

Reply:
We might want to covert tests to import from ts files. That way tests can be compiled to be ESM or CJS based.

Reply:
Spoke offline, this is out of scope for this PR and has been logged as a follow-up task